users-refresher-lambda/src/index.ts

```diff
@@ -12,30 +8,101 @@ import {
 } from "../../shared/panDomainAuth";
 import { p12ToPem } from "./p12ToPem";
 import { getPinboardPermissionOverrides } from "../../shared/permissions";
-import { userTableTTLAttribute } from "../../shared/constants";
 import { getEnvironmentVariableOrThrow } from "../../shared/environmentVariables";
-
-const ONE_DAY_IN_SECONDS = 60 * 60 * 24;
+import { DocumentClient } from "aws-sdk/lib/dynamodb/document_client";
 
 const GUARDIAN_EMAIL_DOMAIN = "@guardian.co.uk";
 
+interface BasicUser {
+  resourceName?: string;
+  email: string;
+  firstName?: string;
+  lastName?: string;
+  isMentionable: boolean;
+}
+
 const S3 = new AWS.S3(standardAwsConfig);
 const dynamo = new AWS.DynamoDB.DocumentClient(standardAwsConfig);
 
 export const handler = async () => {
+  const now = new Date();
   const usersTableName = getEnvironmentVariableOrThrow("usersTableName");
 
+  const getStoredUsers = async (
+    startKey?: DocumentClient.Key
+  ): Promise<BasicUser[]> => {
+    const userResults = await dynamo
+      .scan({
+        TableName: usersTableName,
+        ExclusiveStartKey: startKey,
+        AttributesToGet: ["email", "isMentionable"],
+      })
+      .promise();
+
+    const storedUsers =
+      userResults.Items?.map((user) => user as BasicUser) || [];
+
+    if (userResults.LastEvaluatedKey) {
+      return [
+        ...storedUsers,
+        ...(await getStoredUsers(userResults.LastEvaluatedKey)),
+      ];
+    } else {
+      return storedUsers;
+    }
+  };
+
   const emailsOfUsersWithPinboardPermission = (
     await getPinboardPermissionOverrides(S3)
-  )?.reduce(
+  )?.reduce<string[]>(
     (acc, { userId, active }) => (active ? [...acc, userId] : acc),
-    [] as string[]
+    []
   );
 
   if (!emailsOfUsersWithPinboardPermission) {
     throw Error("Could not get list of users with 'pinboard' permission.");
   }
 
+  const isNightlyFullRun = now.getHours() === 0 && now.getMinutes() === 0;
```

Comment:
Would it be worth adding a parameter to allow manually running a full run in case of errors/emergencies?

Reply:
yeah I actually was thinking about something like that. Perhaps two schedules...

1. every minute, sends some `minimal: true` flag in the payload
2. every 24 hours, no payload (i.e. same as a manual run)

Reply:
done in https://github.com/guardian/editorial-tools-pinboard/pull/123/commits/c4c5d6747d89da353bc9c3b746e0572f8fcd54cd